src/vcf-autosuggest.js

```diff
@@ -508,49 +510,47 @@ import './vcf-autosuggest-overlay';
                 }
             })
         );
+        this._changeTextFieldValue(optLbl);
         if(!keepDropdownOpened) {
-            this._changeTextFieldValue(optLbl);
             this.opened = false;
-            this.$.textField.blur();
-        } else {
-            this._changeTextFieldValue(optLbl);
-            this._inputValueChanged(optLbl);
+            this._textFieldFocus(false)
         }
     }
 
     clear(keepDropdownOpened=false) {
         if(!keepDropdownOpened) this._applyValue(this._hasDefaultOption() ? this._defaultOption.key : '', true);
-        this.$.textField.focus();
+        this._textFieldFocus();
         if(!keepDropdownOpened) {
             this.opened = false;
-            this.$.textField.blur();
+            this._textFieldFocus(false);
         }
     }
 
     _changeTextFieldValue(newValue) {
-        this.$.textField.value = newValue;
-
-        this.$.textField.dispatchEvent(
-            new Event('input', {
-                bubbles: true,
-                cancelable: true
-            })
-        );
-
-        this.$.textField.dispatchEvent(
-            new Event('value-changed', {
-                bubbles: true,
-                cancelable: true
-            })
-        );
-
-        this.$.textField.dispatchEvent(
-            new Event('change', {
-                bubbles: true,
-                cancelable: true
-            })
-        );
-
+        if(typeof this.$ !== 'undefined') {
```

Comment:
All this is related to check if "ready"has already been executed, again to avoid console errors